app/src/main/java/space/taran/arknavigator/mvp/model/repo/preview/PreviewGenerators.kt

```diff
@@ -45,26 +47,64 @@ object PreviewGenerators {
             Log.d(PREVIEWS, "Thumbnail for image $path generated in $time1 ms")
             return
         }
+
         val ext = extension(path)
         generatorsByExt[ext]?.let { generator ->
             val time2 = measureTimeMillis {
-                val thumbnail = if (ext == "txt") {
-                    resizePreviewToThumbnail(generator(path))
-                } else {
-                    val preview = generator(path)
-                    storePreview(previewPath, preview)
-                    resizePreviewToThumbnail(preview)
-                }
+                val preview = generator(path)
+                storePreview(previewPath, preview)
+                val thumbnail = resizePreviewToThumbnail(preview)
                 storeThumbnail(thumbnailPath, thumbnail)
             }
             Log.d(
                 PREVIEWS,
                 "Preview and thumbnail generated for $path in $time2 ms"
             )
+            return
         } ?: Log.d(
             PREVIEWS,
             "No generators found for type .${extension(path)} ($path)"
         )
+        // This section called when the code will failed to generate preview for
+        // pdf, link, txt extension.This code will match with existing factories'
+        // acceptedExtensions as well as acceptedMimeTypes (if ext is blank).
+        if (PlainTextKindFactory.isValid(path)) {
+            generatorsByExt["txt"]?.let { generator ->
```

Comment:
If you need to use this generator directly, just initialize it separately, save it both to map and variable and use the variable here. Don't use String constants for accessing it.

Reply:
Resolved.